b7cd118... config_office/configure: Regenerate.  Fixes naive builds.
d130ba7... Wordcount: minor HIG tweaks.
01a6fce... Zoom: minor HIG spacing tweaks.
67f2b82... Zoom: fix align-test and update zoom.xml.
fa8d7ec... Zoom: add align-test.xml to illustrate zoom alignment bug.
6c70e47... Find-and-Replace: request font-style-name="Bold", does not seem to work.
afacef2... Find-and-Replace: Add alignment for combobox labels.
27de12f... Find-and-Replace: Replace pairs of <flow> with <align>. Fixes ugly non-emptiness of more-<hbox>.
5e1f71e... Find-and-Replace: add nice 12pt border.
ba667ea... Find-and-Replace: also add rowsep at lower fixedline.
150333c... Find-and-Replace: Add poor man's colsep and rowsep substitute spaces.
fe770b8... Find-and-Replace: use <vbox cnt:x-expand="false"> for buttons.
ab48d2a... Find-and-Replace: add <vbox>s and use cnt:expand="false" to stop vertical button expansion.
099e60d... Find-and-Replace: use sc icon. Fixes FIND button height.
a35e707... Bugfix have empty/invisible containers not take up space.
5520d73... Find-and-replace: Fix/remove most property errors.
ae26364... Table: make x, y expand default.
4269438... Use OSL_TRACE rather than printf. Fixes warning visibility.
3f9b304... Boxtest: add <vbox><table x-expand="true"> pair to trigger property errors.
bc138f1... Junk extra <table> from boxtest. Fixes visual output.
0cf4adf... Eradicate PROPHELPER_SET_INFO: junk static caching.
f10be27... Update awk+sed localize generation rules. Fixes multiple (wrong) titles in Find & Replace.
5f02d6f... Use dash (-) instead of underscore (_) to separate language modifiers. Fixes en_GB etc.
e765573... Add "defbutton=true" to dialogs. Fixes ENTER doing default action: n#439222.
284c473... Michael's new table-based find and replace.
1c1f4cc... Apply Michael's boxtest patch.


diff --git sc/uiconfig/layout/insert-sheet.xml sc/uiconfig/layout/insert-sheet.xml
index 0ce5843..517a24d 100644
--- sc/uiconfig/layout/insert-sheet.xml
+++ sc/uiconfig/layout/insert-sheet.xml
@@ -49,7 +49,7 @@
 	    </vbox>
 	</hbox>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK"/>
+            <okbutton defbutton="true" id="BTN_OK"/>
             <cancelbutton id="BTN_CANCEL"/>
             <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git sc/uiconfig/layout/makefile.mk sc/uiconfig/layout/makefile.mk
index 4688e04..4832e5c 100644
--- sc/uiconfig/layout/makefile.mk
+++ sc/uiconfig/layout/makefile.mk
@@ -60,8 +60,8 @@ all .PHONY:
 .INCLUDE :  target.mk
 
 localize.sdf: $(PRJ)/source/ui/miscdlgs/localize.sdf $(PRJ)/source/ui/src/localize.sdf
-	grep instbdlg.src $(PRJ)/source/ui/miscdlgs/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sc\\uiconfig\\layout\\insert-sheet.xml", $$3, "layout", $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\t_label/\tRID_SCDLG_INSERT_TABLE_title/' > insert-sheet-$@
-	grep -E 'miscdlgs.src.*(FT_DEST|FT_INSERT|STR_NEWDOC|RID_SCDLG_MOVETAB|BTN_COPY)' $(PRJ)/source/ui/src/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sc\\uiconfig\\layout\\move-copy-sheet.xml", $$3, "layout", $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\t_label/\tRID_SCDLG_MOVETAB_title/' > move-copy-sheet-$@
-	grep sortdlg.src $(PRJ)/source/ui/src/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sc\\uiconfig\\layout\\sort-options.xml", $$3, "layout", $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\t_label/\tRID_SCDLG_SORT_title/' > sort-options-$@
+	grep instbdlg.src $(PRJ)/source/ui/miscdlgs/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sc\\uiconfig\\layout\\insert-sheet.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SCDLG_INSERT_TABLE/\t/' -e 's/\t_label/\tRID_SCDLG_INSERT_TABLE_title/' > insert-sheet-$@
+	grep -E 'miscdlgs.src.*(FT_DEST|FT_INSERT|STR_NEWDOC|RID_SCDLG_MOVETAB|BTN_COPY)' $(PRJ)/source/ui/src/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sc\\uiconfig\\layout\\move-copy-sheet.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SCDLG_MOVETAB/\t/' -e 's/\t_label/\tRID_SCDLG_MOVETAB_title/'> move-copy-sheet-$@
+	grep sortdlg.src $(PRJ)/source/ui/src/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sc\\uiconfig\\layout\\sort-options.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SCDLG_SORT/\t/' -e 's/\t_label/\tRID_SCDLG_SORT_title/' -e 's/\tRID_SCPAGE_SORT_OPTIONS/\t/' -e 's/\tRID_SCPAGE_SORT_FIELDS/\t/' > sort-options-$@
 	echo '#empty' | cat - insert-sheet-$@ move-copy-sheet-$@ sort-options-$@ > $@
 	rm -f *-$@
diff --git sc/uiconfig/layout/move-copy-sheet.xml sc/uiconfig/layout/move-copy-sheet.xml
index 79720df..6e68b5a 100644
--- sc/uiconfig/layout/move-copy-sheet.xml
+++ sc/uiconfig/layout/move-copy-sheet.xml
@@ -15,7 +15,7 @@
 	    <flow/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
diff --git sc/uiconfig/layout/string-input.xml sc/uiconfig/layout/string-input.xml
index 0a59899..a29a564 100644
--- sc/uiconfig/layout/string-input.xml
+++ sc/uiconfig/layout/string-input.xml
@@ -8,7 +8,7 @@
         <fixedtext id="FT_LABEL" label="FT-LABEL"/>
         <edit has_border="true" id="ED_INPUT" tabstop="true"/>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
 	</dialogbuttonhbox>
diff --git svx/uiconfig/layout/find-and-replace.xml svx/uiconfig/layout/find-and-replace.xml
index ad5af2b..aff0531 100644
--- svx/uiconfig/layout/find-and-replace.xml
+++ svx/uiconfig/layout/find-and-replace.xml
@@ -3,109 +3,164 @@
      i18n translated XML files are generated from this template by
      transex3/layout/tralay.  !-->
 
-<sfxmodelessdialog _title="Find &amp; Replace" help-id="SID_SEARCH_DLG" closeable="true" sizeable="true" moveable="true" output-size="true" id="RID_SVXDLG_SEARCH" sv-look="true" xmlns="http://openoffice.org/2007/layout" xmlns:cnt="http://openoffice.org/2007/layout/container">
-    <vbox spacing="5" border="5">
+<sfxmodelessdialog _title="Find &amp; Replace" help-id="SID_SEARCH_DLG" closeable="true" sizeable="true" moveable="true" id="RID_SVXDLG_SEARCH" xmlns="http://openoffice.org/2007/layout" xmlns:cnt="http://openoffice.org/2007/layout/container">
+    <hbox border="12">
+    <table columns="3">
 	<string id="RID_SVXSTR_SEARCH_STYLES" _text="Including Styles"/>
 	<string id="RID_SVXSTR_SEARCH" _text="(Search)"/>
 	<string id="RID_SVXSTR_REPLACE" _text="(Replace)"/>
 	<string id="STR_WORDCALC" _text="~Entire cells"/>
 	<string id="STR_MORE_BTN" _text="More ~Options"/>
 	<string id="STR_LESS_BTN" _text="Less ~Options"/>
-	<hbox homogeneous="false" spacing="5" border="0">
-	    <vbox spacing="5" border="0">
-		<fixedtext id="FT_SEARCH" _label="~Search for"/>
-		<combobox dropdown="true" id="ED_SEARCH" tabstop="true" cnt:expand="false"/>
-		<listbox has_border="true" dropdown="true" show="false" id="LB_SEARCH" sort="true" tabstop="true"/>
-		<fixedtext show="false" id="FT_SEARCH_ATTR" word-break="true"/>
-		<fixedtext show="false" id="FT_SEARCH_FORMATS" word-break="true"/>
-		<flow/>
-		<flow/>
+
+	<!-- row -->
+	<align valign="1" halign="0">
+	    <fixedtext id="FT_SEARCH" _label="~Search for" font-style-name="Bold"/>
+	</align>
+	<!-- poor man's colsep -->
+	<min-size cnt:x-expand="false" min-width="12"/>
+	<vbox cnt:x-expand="false">
+	    <imagebutton defbutton="true" id="BTN_SEARCH" tabstop="true" _label="~Find" cnt:expand="false"
+			 image-align="0" graphic="res/commandimagelist/sc_searchdialog.png"/>
+	</vbox>
+	<!-- row: poor man's rowsep -->
+	<min-size cnt:col-span="3" cnt:x-expand="false" min-height="12"/>
+	<!-- row -->
+	<vbox spacing="0" border="0" cnt:x-expand="true">
+	    <combobox dropdown="true" id="ED_SEARCH" tabstop="true" cnt:expand="true"/>
+	    <listbox has_border="true" dropdown="true" show="false" id="LB_SEARCH" sort="true" tabstop="true" cnt:expand="true"/>
+	</vbox>
+	<!-- poor man's colsep -->
+	<min-size cnt:x-expand="false" min-width="12"/>
+	<vbox cnt:x-expand="false">
+	    <pushbutton id="BTN_SEARCH_ALL" tabstop="true" _label="Find ~All" cnt:expand="false"/>
+	</vbox>
+	<!-- row -->
+	<vbox spacing="0" border="0" cnt:expand="false" cnt:col-span="3">
+	    <fixedtext show="false" id="FT_SEARCH_ATTR" word-break="true"/>
+	    <fixedtext show="false" id="FT_SEARCH_FORMATS" word-break="true"/>
+	</vbox>
+	<!-- row: poor man's rowsep -->
+	<min-size cnt:col-span="3" cnt:x-expand="false" min-height="6"/>
+	<!-- row -->
+	<fixedline cnt:y-expand="false" cnt:col-span="3"/>
+	<!-- row: poor man's rowsep -->
+	<min-size cnt:col-span="3" cnt:x-expand="false" min-height="6"/>
+	<!-- row -->
+	<align valign="1" halign="0">
+	    <fixedtext id="FT_REPLACE" _label="Re~place with" font-style-name="Bold"/>
+	</align>
+	<!-- poor man's colsep -->
+	<min-size cnt:x-expand="false" min-width="12"/>
+	<vbox cnt:x-expand="false">
+	    <pushbutton id="BTN_REPLACE" tabstop="true" _label="~Replace" cnt:expand="false"/>
+	</vbox>
+	<!-- row: poor man's rowsep -->
+	<min-size cnt:col-span="3" cnt:x-expand="false" min-height="12"/>
+	<!-- row -->
+	<vbox spacing="0" border="0" cnt:x-expand="true">
+	    <combobox dropdown="true" id="ED_REPLACE" tabstop="true" cnt:expand="true"/>
+	    <listbox has_border="true" dropdown="true" show="false" id="LB_REPLACE" sort="true" tabstop="true" cnt:expand="true"/>
+	</vbox>
+	<!-- poor man's colsep -->
+	<min-size cnt:x-expand="false" min-width="12"/>
+	<vbox cnt:x-expand="false">
+	    <pushbutton id="BTN_REPLACE_ALL" tabstop="true" _label="Replace A~ll" cnt:expand="false"/>
+	</vbox>
+	<!-- row -->
+	<vbox spacing="6" border="0" cnt:col-span="3">
+	    <fixedtext show="false" id="FT_REPLACE_ATTR" word-break="true"/>
+	    <fixedtext show="false" id="FT_REPLACE_FORMATS" word-break="true"/>
+	    <fixedline show="false" id="FL_SEARCH_COMPONENT"/>
+	</vbox>
+	<!-- row: poor man's rowsep -->
+	<min-size cnt:col-span="3" cnt:x-expand="false" min-height="12"/>
+	<!-- row -->
+	<vbox spacing="6" border="0" cnt:col-span="3">
+	    <hbox spacing="12">
+                <checkbox id="CB_MATCH_CASE" tabstop="true" _label="Ma~tch case"/>
+		<min-size min-width="0"/>
+		<vbox cnt:x-expand="false" cnt:padding="12">
+                    <pushbutton show="false" id="BTN_COMPONENT_1" tabstop="true" cnt:expand="false" cnt:padding="12"/>
+		</vbox>
+                <checkbox id="CB_WHOLE_WORDS" tabstop="true" _label="Whole wor~ds only"/>
+		<min-size min-width="0"/>
+		<vbox cnt:x-expand="false" cnt:padding="12">
+                    <pushbutton show="false" id="BTN_COMPONENT_2" tabstop="true" cnt:expand="false"/>
+		</vbox>
 		<flow/>
+	    </hbox>
+	</vbox>
+	<!-- row: poor man's rowsep -->
+	<min-size cnt:col-span="3" cnt:x-expand="false" min-height="6"/>
+	<!-- row -->
+        <fixedline cnt:y-expand="false" cnt:col-span="3"/>
+	<!-- row: poor man's rowsep -->
+	<min-size cnt:col-span="3" cnt:x-expand="false" min-height="6"/>
+	<!-- row -->
+	<hbox cnt:col-span="3">
+	    <vbox cnt:expand="false">
+		<morebutton delta="174" map-unit="MAP_APPFONT" id="BTN_MORE" state="false" tabstop="true" cnt:expand="false"/>
 	    </vbox>
-	    <vbox cnt:expand="false" spacing="5" border="0">
-		<min-size min-width="100" cnt:expand="false"/>
-		<pushbutton default="true" id="BTN_SEARCH" tabstop="true" _label="~Find" cnt:expand="false"/>
-		<pushbutton id="BTN_SEARCH_ALL" tabstop="true" _label="Find ~All" cnt:expand="false"/>
-	    </vbox>
-	</hbox>
-        <hbox spacing="5" border="0">
 	    <flow/>
-	    <vbox cnt:expand="false" spacing="5" border="0">
-		<min-size min-width="100" cnt:expand="false"/>
-		<fixedline id="FL_SEARCH_COMMAND"/>
-	    </vbox>
-	</hbox>
-	<hbox homogeneous="false" spacing="5" border="0">
-	    <vbox spacing="5" border="0">
-		<fixedtext id="FT_REPLACE" _label="Re~place with"/>
-		<combobox dropdown="true" id="ED_REPLACE" tabstop="true" cnt:expand="false"/>
-		<listbox has_border="true" dropdown="true" show="false" id="LB_REPLACE" sort="true" tabstop="true"/>
-		<fixedtext show="false" id="FT_REPLACE_ATTR" word-break="true"/>
-		<fixedtext show="false" id="FT_REPLACE_FORMATS" word-break="true"/>
-		<fixedline show="false" id="FL_SEARCH_COMPONENT"/>
-		<flow/>
-		<flow/>
-		<flow/>
-	    </vbox>
-	    <vbox cnt:expand="false" spacing="5" border="0">
-		<min-size min-width="100" cnt:expand="false"/>
-		<pushbutton id="BTN_REPLACE" tabstop="true" _label="~Replace" cnt:expand="false"/>
-		<pushbutton id="BTN_REPLACE_ALL" tabstop="true" _label="Replace A~ll" cnt:expand="false"/>
-	    </vbox>
-	</hbox>
-        <checkbox id="CB_MATCH_CASE" tabstop="true" _label="Ma~tch case"/>
-        <pushbutton show="false" id="BTN_COMPONENT_1" tabstop="true" cnt:expand="false"/>
-        <checkbox id="CB_WHOLE_WORDS" tabstop="true" _label="Whole wor~ds only"/>
-        <pushbutton show="false" id="BTN_COMPONENT_2" tabstop="true" cnt:expand="false"/>
-        <fixedline id="FL_BUTTONS"/>
-	<hbox>
-            <morebutton delta="174" map-unit="MAP_APPFONT" id="BTN_MORE" state="false" tabstop="true"/>
-            <dialogbuttonhbox>
-		<helpbutton id="BTN_HELP" tabstop="true" cnt:expand="false"/>
-		<cancelbutton id="BTN_CLOSE" tabstop="true" _label="~Close" cnt:expand="false"/>
+            <dialogbuttonhbox cnt:expand="false">
+		<vbox cnt:expand="false" cnt:fill="false">
+		    <helpbutton id="BTN_HELP" tabstop="true" cnt:expand="false"/>
+		</vbox>
+		<vbox cnt:expand="false" cnt:fill="false">
+		    <cancelbutton id="BTN_CLOSE" tabstop="true" _label="~Close" cnt:expand="false"/>
+		</vbox>
             </dialogbuttonhbox>
 	</hbox>
-        <fixedline show="false" id="FL_OPTIONS"/>
-	<hbox homogeneous="false" spacing="5" border="0">
-	    <vbox spacing="5" border="0">
+	<!-- row -->
+        <fixedline show="false" id="FL_OPTIONS" cnt:col-span="3"/>
+	<!-- row -->
+	<hbox spacing="6" border="0" cnt:col-span="3">
+	    <vbox spacing="6" border="0">
+		<min-size min-height="0"/>
 		<checkbox show="false" id="CB_SELECTIONS" tabstop="true" _label="Current selection ~only"/>
 		<checkbox show="false" id="CB_BACKWARDS" tabstop="true" _label="Bac~kwards"/>
 		<checkbox show="false" id="CB_REGEXP" tabstop="true" _label="Regular e~xpressions"/>
 		<hbox>
 		    <checkbox show="false" id="CB_SIMILARITY" tabstop="true" _label="S~imilarity search"/>
-		    <flow/>
-		    <pushbutton show="false" id="PB_SIMILARITY" tabstop="true" _label="~..." cnt:expand="false"/>
-		    <flow/>
+		    <align halign="1">
+		    <vbox cnt:x-expand="false">
+			<pushbutton show="false" id="PB_SIMILARITY" tabstop="true" _label="~..." cnt:expand="false"/>
+		    </vbox>
+		    </align>
 		</hbox>
 		<checkbox show="false" id="CB_LAYOUTS" tabstop="true" _label="Search for St~yles"/>
 		<checkbox show="false" id="CB_JAP_MATCH_FULL_HALF_WIDTH" tabstop="true" _label="Match character width"/>
 		<hbox>
 		    <checkbox show="false" id="CB_JAP_SOUNDS_LIKE" tabstop="true" _label="Sounds like (Japanese)"/>
-		    <flow/>
-		    <pushbutton show="false" id="PB_JAP_OPTIONS" tabstop="true" _label="..." cnt:expand="false"/>
-		    <flow/>
+		    <align halign="1">
+		    <vbox cnt:x-expand="false">
+			<pushbutton show="false" id="PB_JAP_OPTIONS" tabstop="true" _label="..." cnt:expand="false"/>
+		    </vbox>
+		    </align>
 		</hbox>
 
 		<fixedline show="false" id="FL_CALC"/>
-		<hbox spacing="5" border="0">
+		<hbox spacing="6" border="0">
 		    <fixedtext show="false" id="FT_CALC_SEARCHIN" _label="~Search in"/>
 		    <listbox has_border="true" dropdown="true" show="false" id="LB_CALC_SEARCHIN" tabstop="true"/>
 		</hbox>
-		<hbox spacing="5" border="0">
+		<hbox spacing="6" border="0">
 		    <fixedtext show="false" id="FT_CALC_SEARCHDIR" _label="~Search direction"/>
-		    <vbox spacing="5" border="5">
+		    <vbox spacing="6" border="6">
 			<radiobutton show="false" id="RB_CALC_ROWS" _label="Ro~ws"/>
 			<radiobutton show="false" id="RB_CALC_COLUMNS" _label="Col~umns"/>
 		    </vbox>
 		</hbox>
 		<checkbox show="false" id="CB_ALL_SHEETS" _label="Search in all sheets"/>
 	    </vbox>
-	    <vbox cnt:expand="false" spacing="5" border="0">
+	    <vbox spacing="6" border="0" cnt:x-expand="false">
 		<min-size min-width="100" cnt:expand="false"/>
 		<pushbutton show="false" id="BTN_ATTRIBUTE" tabstop="true" _label="Attri~butes..." cnt:expand="false"/>
 		<pushbutton show="false" id="BTN_FORMAT" tabstop="true" _label="For~mat..." cnt:expand="false"/>
 		<pushbutton show="false" id="BTN_NOFORMAT" tabstop="true" _label="~No Format" cnt:expand="false"/>
 	    </vbox>
 	</hbox>
-    </vbox>
+    </table>
+    </hbox>
 </sfxmodelessdialog>
diff --git svx/uiconfig/layout/makefile.mk svx/uiconfig/layout/makefile.mk
index 871db2f..26679e2 100644
--- svx/uiconfig/layout/makefile.mk
+++ svx/uiconfig/layout/makefile.mk
@@ -54,7 +54,7 @@ all .PHONY:
 .INCLUDE :  target.mk
 
 localize.sdf: $(PRJ)/source/dialog/localize.sdf
-	grep srchdlg.src $(PRJ)/source/dialog/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "svx\\uiconfig\\layout\\find-and-replace.xml", $$3, "layout", $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\t_label/\tRID_SVXDLG_SEARCH_title/' > find-and-replace-$@
-	grep zoom.src $(PRJ)/source/dialog/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "svx\\uiconfig\\layout\\zoom.xml", $$3, "layout", $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\t_label/\tRID_SVXDLG_ZOOM_title/' > zoom-$@
+	grep srchdlg.src $(PRJ)/source/dialog/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "svx\\uiconfig\\layout\\find-and-replace.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SVXDLG_SEARCH/\t/' -e 's/\t_label/\tRID_SVXDLG_SEARCH_title/' > find-and-replace-$@
+	grep zoom.src $(PRJ)/source/dialog/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "svx\\uiconfig\\layout\\zoom.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\tRID_SVXDLG_ZOOM/\t/' -e 's/\t_label/\tRID_SVXDLG_ZOOM_title/' > zoom-$@
 	echo '#empty' | cat - find-and-replace-$@ zoom-$@ > $@
 	rm -f *-$@
diff --git svx/uiconfig/layout/zoom.xml svx/uiconfig/layout/zoom.xml
index 5938643..4b4270f 100644
--- svx/uiconfig/layout/zoom.xml
+++ svx/uiconfig/layout/zoom.xml
@@ -8,10 +8,10 @@
              id="RID_SVXDLG_ZOOM" _title="Zoom &amp; View Layout" optimumsize="true"
 	     help-id="SID_ATTR_ZOOM"
              has_border="true" sizeable="true" moveable="true" closeable="true">
-    <vbox spacing="5" border="5">
+    <vbox spacing="0" border="12">
 	<hbox spacing="0" border="0">
-	    <vbox spacing="5" border="5">
-		<fixedline id="FL_ZOOM" _text="Zoom factor" cnt:expand="true"/>
+	    <vbox spacing="6" border="0">
+		<fixedline id="FL_ZOOM" _text="Zoom factor" cnt:expand="false"/>
 		<radiobutton radiogroup="zoom-factor" id="BTN_OPTIMAL"   _label="Optimal"/>
 		<radiobutton radiogroup="zoom-factor" id="BTN_WHOLE_PAGE" _label="Whole Page"/>
 		<radiobutton radiogroup="zoom-factor" id="BTN_PAGE_WIDTH" _label="Page Width"/>
@@ -20,7 +20,7 @@
 		    <align cnt:expand="false" cnt:fill="true">
 			<radiobutton cnt:v-align="0.5" cnt:v-fill="0" radiogroup="zoom-factor" id="BTN_USER" _label="Variable"/>
 		    </align>
-		    <flow cnt:padding="10" cnt:expand="false"/>
+		    <flow cnt:padding="12" cnt:expand="false"/>
 		    <metricfield id="ED_USER" value-step="1"
 				 repeat="true" has_border="true" spin="true"
 				 _text="100%" unit="11" custom-unit-text="%"
@@ -28,15 +28,16 @@
 				 cnt:expand="false"/>
 		</hbox>
 	    </vbox>
-	    <vbox spacing="5" border="5">
-		<fixedline id="FL_VIEWLAYOUT" _text="View layout" cnt:expand="true"/>
+	    <flow/>
+	    <vbox spacing="6" border="0">
+		<fixedline id="FL_VIEWLAYOUT" _text="View layout" cnt:expand="false"/>
 		<radiobutton radiogroup="zoom-layout" id="BTN_AUTOMATIC" _label="~Automatic"/>
 		<radiobutton radiogroup="zoom-layout" id="BTN_SINGLE"    _label="~Single page"/>
 		<hbox cnt:expand="false" cnt:fill="true">
 		    <align cnt:expand="false" cnt:fill="true">
 			<radiobutton cnt:v-align="0.5" cnt:v-fill="0" radiogroup="zoom-layout" id="BTN_COLUMNS" _label="~Columns"/>
 		    </align>
-		    <flow cnt:padding="10" cnt:expand="false"/>
+		    <flow cnt:padding="12" cnt:expand="false"/>
 		    <metricfield id="ED_COLUMNS" value-step="1"
 				 repeat="true" has_border="true" spin="true"
 				 _text="100%"
@@ -45,15 +46,15 @@
 				 spin-value="1"
 				 cnt:expand="false"/>
 		</hbox>
-		<hbox><flow cnt:expand="false" cnt:padding="10"/><checkbox id="CHK_BOOK" _label="~Book mode"/></hbox>
+		<hbox><flow cnt:expand="false" cnt:padding="12"/><checkbox id="CHK_BOOK" _label="~Book mode"/></hbox>
 	    </vbox>
 	</hbox>
-        <fixedline cnt:padding="1" id="FL_BOTTOM"/>
-	<dialogbuttonhbox border="5" spacing="5">
+        <fixedline cnt:padding="6" id="FL_BOTTOM"/>
+	<dialogbuttonhbox border="0" spacing="12">
 	    <flow/>
-	    <okbutton     id="BTN_ZOOM_OK"/>
+	    <okbutton defbutton="true" id="BTN_ZOOM_OK"/>
 	    <cancelbutton id="BTN_ZOOM_CANCEL"/>
-	    <helpbutton   id="BTN_ZOOM_HELP"/>
+	    <helpbutton id="BTN_ZOOM_HELP"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
diff --git sw/uiconfig/layout/makefile.mk sw/uiconfig/layout/makefile.mk
index 5f4bc52..8debaa6 100644
--- sw/uiconfig/layout/makefile.mk
+++ sw/uiconfig/layout/makefile.mk
@@ -50,6 +50,7 @@ all .PHONY:
 .INCLUDE :  target.mk
 
 localize.sdf: $(PRJ)/source/ui/dialog/localize.sdf
-	grep wordcountdialog.src $(PRJ)/source/ui/dialog/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sw\\uiconfig\\layout\\wordcount.xml", $$3, "layout", $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\t_label/\tDLG_WORDCOUNT_title/' > wordcount-$@
+#	grep wordcountdialog.src $(PRJ)/source/ui/dialog/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sw\\uiconfig\\layout\\wordcount.xml", $$3, "layout", $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/' -e 's/\t_label/\tDLG_WORDCOUNT_title/' > wordcount-$@
+	grep wordcountdialog.src $(PRJ)/source/ui/dialog/localize.sdf | awk -F'\t' '{{printf "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", "layout", "sw\\uiconfig\\layout\\wordcount.xml", $$3, "layout", $$5 $$6 "_label", "", "", $$8, "0", $$10, $$11, $$12, "", $$14, $$15}}' | sed -e 's/\(\(FL\|STR\)_[^\t]*\)_label/\1_text/'  -e 's/\tDLG_WORDCOUNT/\t/' -e 's/\t_label/\tDLG_WORDCOUNT_title/' > wordcount-$@
 	echo '#empty' | cat - wordcount-$@ > $@
 	rm -f *-$@
diff --git sw/uiconfig/layout/wordcount.xml sw/uiconfig/layout/wordcount.xml
index e922294..fc7ff94 100644
--- sw/uiconfig/layout/wordcount.xml
+++ sw/uiconfig/layout/wordcount.xml
@@ -9,10 +9,10 @@
              id="DLG_WORDCOUNT"
              sizeable="true" moveable="true" optimumsize="true" closeable="true"
              has_border="true" sv-look="true" _title="Word Count">
-    <vbox border="5" spacing="5">
+    <vbox border="12" spacing="6">
         <fixedline id="FL_CURRENT" _text="Current selection"/>
 	<hbox>
-	    <vbox cnt:padding="13" spacing="5">
+	    <vbox cnt:padding="12" spacing="6">
 		<hbox>
 		    <fixedtext id="FT_CURRENTWORD" _label="Words:"/>
 		    <flow cnt:padding="60"/>
@@ -25,11 +25,11 @@
 		</hbox>
 	    </vbox>
 	</hbox>
-	<!-- if we could do cnt:cnt:padding="-13" here, we would not have to close
+	<!-- if we could do cnt:cnt:padding="-12" here, we would not have to close
 	     and reopen the hbox/vbox thing !-->
         <fixedline id="FL_DOC" _text="Whole document"/>
 	<hbox>
-	    <vbox cnt:padding="13" spacing="5">
+	    <vbox cnt:padding="12" spacing="6">
 		<hbox>
 		    <fixedtext id="FT_DOCWORD" _label="Words:"/>
 		    <fixedtext id="FI_DOCWORD" right="true" _label="000"/>
@@ -41,9 +41,9 @@
 	    </vbox>
 	</hbox>
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
-	<dialogbuttonhbox spacing="5">
+	<dialogbuttonhbox spacing="12">
             <flow/>
-            <okbutton id="PB_OK"/>
+            <okbutton defbutton="true" id="PB_OK"/>
             <helpbutton id="PB_HELP"/>
         </dialogbuttonhbox>
     </vbox>
diff --git toolkit/source/layout/core/bin.cxx toolkit/source/layout/core/bin.cxx
index 62ea068..dad845d 100644
--- toolkit/source/layout/core/bin.cxx
+++ toolkit/source/layout/core/bin.cxx
@@ -165,6 +165,12 @@ Align::allocateArea( const awt::Rectangle &rArea )
     allocateChildAt( mxChild, aChildArea );
 }
 
+bool
+Align::emptyVisible ()
+{
+    return true;
+}
+
 /* MinSize */
 
 MinSize::MinSize() : Bin()
@@ -178,6 +184,12 @@ MinSize::MinSize() : Bin()
              &mnMinHeight );
 }
 
+bool
+MinSize::emptyVisible ()
+{
+    return true;
+}
+
 awt::Size SAL_CALL MinSize::getMinimumSize()
     throw(uno::RuntimeException)
 {
diff --git toolkit/source/layout/core/bin.hxx toolkit/source/layout/core/bin.hxx
index c23c371..0b75bc2 100644
--- toolkit/source/layout/core/bin.hxx
+++ toolkit/source/layout/core/bin.hxx
@@ -75,8 +75,6 @@ public:
     // css::awt::XLayoutConstrains
     virtual css::awt::Size SAL_CALL getMinimumSize()
         throw(css::uno::RuntimeException);
-
-    PROPHELPER_SET_INFO
 };
 
 // Align gives control over child position on the allocated space.
@@ -91,11 +89,11 @@ protected:
 public:
     Align();
 
+    bool emptyVisible ();
+
     // css::awt::XLayoutContainer
     virtual void SAL_CALL allocateArea( const css::awt::Rectangle &rArea )
         throw (css::uno::RuntimeException);
-
-    PROPHELPER_SET_INFO
 };
 
 // Makes child request its or a specified size, whatever is larger.
@@ -108,11 +106,10 @@ protected:
 public:
     MinSize();
 
+    bool emptyVisible ();
     // css::awt::XLayoutContainer
     virtual css::awt::Size SAL_CALL getMinimumSize()
         throw(css::uno::RuntimeException);
-
-    PROPHELPER_SET_INFO
 };
 
 } //  namespace layoutimpl
diff --git toolkit/source/layout/core/box-base.cxx toolkit/source/layout/core/box-base.cxx
index f0b7a42..f03cd5c 100644
--- toolkit/source/layout/core/box-base.cxx
+++ toolkit/source/layout/core/box-base.cxx
@@ -69,9 +69,9 @@ static bool isVisible( uno::Reference< awt::XLayoutConstrains > xWidget )
         uno::Sequence< uno::Reference< awt::XLayoutConstrains > > aChildren
             = xContainer->getChildren();
 
-        // FIXME: <flow> workaround: empty visible containers always visible:
-        if ( !aChildren.getLength() )
-            return true;
+        if (!aChildren.getLength ())
+            if (Container *c = dynamic_cast <Container*> (xWidget.get ()))
+                return c->emptyVisible ();
 
         for ( int i = 0; i < aChildren.getLength(); i++ )
             if ( isVisible( aChildren[i] ) )
diff --git toolkit/source/layout/core/box-base.hxx toolkit/source/layout/core/box-base.hxx
index 4fa7e53..67365a3 100644
--- toolkit/source/layout/core/box-base.hxx
+++ toolkit/source/layout/core/box-base.hxx
@@ -57,7 +57,6 @@ public:
     struct ChildProps: public PropHelper
     {
         //ChildProps( ChildProps* );
-        PROPHELPER_SET_INFO
     };
 
 protected:
diff --git toolkit/source/layout/core/box.hxx toolkit/source/layout/core/box.hxx
index 98190ca..d99a603 100644
--- toolkit/source/layout/core/box.hxx
+++ toolkit/source/layout/core/box.hxx
@@ -83,8 +83,6 @@ public:
     // helper: mix of getMinimumSize() and getHeightForWidth()
     css::awt::Size calculateSize( long nWidth = 0 );
 
-    PROPHELPER_SET_INFO
-
 private:
     /* Helpers to deal with the joint Box directions. */
     inline int primDim (const css::awt::Size &size)
diff --git toolkit/source/layout/core/container.cxx toolkit/source/layout/core/container.cxx
index c9b4513..608d5a4 100644
--- toolkit/source/layout/core/container.cxx
+++ toolkit/source/layout/core/container.cxx
@@ -50,6 +50,12 @@ Container::Container()
     setChangeListener( this );
 }
 
+bool
+Container::emptyVisible ()
+{
+    return false;
+}
+
 uno::Any
 Container::queryInterface( const uno::Type & rType ) throw (uno::RuntimeException)
 {
diff --git toolkit/source/layout/core/container.hxx toolkit/source/layout/core/container.hxx
index 5cac33d..71fba80 100644
--- toolkit/source/layout/core/container.hxx
+++ toolkit/source/layout/core/container.hxx
@@ -73,6 +73,8 @@ public:
     Container();
     virtual ~Container() {}
 
+    virtual bool emptyVisible ();
+
     // XInterface
     virtual void SAL_CALL acquire() throw() { PropHelper::acquire(); }
     virtual void SAL_CALL release() throw() { PropHelper::release(); }
@@ -129,8 +131,6 @@ public:
     css::awt::Size SAL_CALL calcAdjustedSize( const css::awt::Size& rNewSize )
         throw(css::uno::RuntimeException) { return rNewSize; }
 
-PROPHELPER_SET_INFO
-
 protected:
     void propertiesChanged();
 };
diff --git toolkit/source/layout/core/dialogbuttonhbox.cxx toolkit/source/layout/core/dialogbuttonhbox.cxx
index 67a7ec0..0475723 100644
--- toolkit/source/layout/core/dialogbuttonhbox.cxx
+++ toolkit/source/layout/core/dialogbuttonhbox.cxx
@@ -39,13 +39,12 @@
 #include "proplist.hxx"
 
 #if TEST_LAYOUT && !defined( DBG_UTIL )
-#include <cstdio>
 #undef DBG_ERROR
-#define DBG_ERROR printf
+#define DBG_ERROR OSL_TRACE
 #undef DBG_ERROR1
-#define DBG_ERROR1 printf
+#define DBG_ERROR1 OSL_TRACE
 #undef DBG_ERROR2
-#define DBG_ERROR2 printf
+#define DBG_ERROR2 OSL_TRACE
 #endif /* TEST_LAYOUT && !DBG_UTIL */
 
 namespace layoutimpl
diff --git toolkit/source/layout/core/flow.cxx toolkit/source/layout/core/flow.cxx
index 4bfd1b9..b004336 100644
--- toolkit/source/layout/core/flow.cxx
+++ toolkit/source/layout/core/flow.cxx
@@ -56,6 +56,12 @@ Flow::Flow()
              &mnSpacing );
 }
 
+bool
+Flow::emptyVisible ()
+{
+    return true;
+}
+
 void SAL_CALL
 Flow::addChild( const uno::Reference< awt::XLayoutConstrains >& xChild )
     throw (uno::RuntimeException, css::awt::MaxChildrenException)
diff --git toolkit/source/layout/core/flow.hxx toolkit/source/layout/core/flow.hxx
index d550947..91aa829 100644
--- toolkit/source/layout/core/flow.hxx
+++ toolkit/source/layout/core/flow.hxx
@@ -63,6 +63,8 @@ protected:
 public:
     Flow();
 
+    bool emptyVisible ();
+
     // css::awt::XLayoutContainer
     virtual void SAL_CALL addChild( const css::uno::Reference< css::awt::XLayoutConstrains >& Child )
         throw (css::uno::RuntimeException, css::awt::MaxChildrenException);
@@ -87,8 +89,6 @@ public:
     virtual sal_Int32 SAL_CALL getHeightForWidth( sal_Int32 nWidth )
         throw(css::uno::RuntimeException);
 
-PROPHELPER_SET_INFO
-
 private:
     // shared between getMinimumSize() and getHeightForWidth()
     css::awt::Size calculateSize( long nMaxWidth );
diff --git toolkit/source/layout/core/helper.cxx toolkit/source/layout/core/helper.cxx
index ec4f80f..ff13269 100644
--- toolkit/source/layout/core/helper.cxx
+++ toolkit/source/layout/core/helper.cxx
@@ -42,13 +42,12 @@
 #include "proplist.hxx"
 
 #if TEST_LAYOUT && !defined( DBG_UTIL )
-#include <cstdio>
 #undef DBG_ERROR
-#define DBG_ERROR printf
+#define DBG_ERROR OSL_TRACE
 #undef DBG_ERROR1
-#define DBG_ERROR1 printf
+#define DBG_ERROR1 OSL_TRACE
 #undef DBG_ERROR2
-#define DBG_ERROR2 printf
+#define DBG_ERROR2 OSL_TRACE
 #endif /* TEST_LAYOUT && !DBG_UTIL */
 
 namespace layoutimpl
@@ -357,6 +356,11 @@ PropHelper::queryInterface( const ::com::sun::star::uno::Type & rType ) throw(::
     return OPropertySetHelper::queryInterface( rType );
 }
 
+uno::Reference <beans::XPropertySetInfo> SAL_CALL PropHelper::getPropertySetInfo () throw (uno::RuntimeException)
+{
+    return css::uno::Reference <css::beans::XPropertySetInfo> (createPropertySetInfo (getInfoHelper ()));
+}
+
 } // namespace layoutimpl
 
 #include <awt/vclxbutton.hxx>
diff --git toolkit/source/layout/core/helper.hxx toolkit/source/layout/core/helper.hxx
index 806efd0..bc74690 100644
--- toolkit/source/layout/core/helper.hxx
+++ toolkit/source/layout/core/helper.hxx
@@ -106,23 +106,13 @@ public:
     virtual void SAL_CALL getFastPropertyValue( css::uno::Any& rValue,
                                                 sal_Int32 nHandle ) const;
 
-    // you -must- use this macro in sub-classes that define new properties.
-    // NB. 'static' ...
-    // com::sun::star::beans::XMultiPropertySet
-#define PROPHELPER_SET_INFO \
-    css::uno::Reference< css::beans::XPropertySetInfo > SAL_CALL \
-    getPropertySetInfo() throw(css::uno::RuntimeException) \
-    { \
-        static css::uno::Reference< css::beans::XPropertySetInfo > xInfo( \
-            createPropertySetInfo( getInfoHelper() ) ); \
-        return xInfo; \
-    }
-    PROPHELPER_SET_INFO
+    virtual css::uno::Reference <css::beans::XPropertySetInfo> SAL_CALL getPropertySetInfo () throw (css::uno::RuntimeException);
 
     struct Listener
     {
         virtual void propertiesChanged() = 0;
     };
+
     void setChangeListener( Listener *pListener )
     {
         mpListener = pListener;
diff --git toolkit/source/layout/core/proplist.cxx toolkit/source/layout/core/proplist.cxx
index 857bb51..483730b 100644
--- toolkit/source/layout/core/proplist.cxx
+++ toolkit/source/layout/core/proplist.cxx
@@ -42,13 +42,12 @@
 #include "helper.hxx"
 
 #if TEST_LAYOUT && !defined( DBG_UTIL )
-#include <cstdio>
 #undef DBG_ERROR
-#define DBG_ERROR printf
+#define DBG_ERROR OSL_TRACE
 #undef DBG_ERROR1
-#define DBG_ERROR1 printf
+#define DBG_ERROR1 OSL_TRACE
 #undef DBG_ERROR2
-#define DBG_ERROR2 printf
+#define DBG_ERROR2 OSL_TRACE
 #endif /* TEST_LAYOUT && !DBG_UTIL */
 
 namespace layoutimpl
diff --git toolkit/source/layout/core/root.cxx toolkit/source/layout/core/root.cxx
index bb264f3..eafafca 100644
--- toolkit/source/layout/core/root.cxx
+++ toolkit/source/layout/core/root.cxx
@@ -32,7 +32,6 @@
 #include "root.hxx"
 
 #include <cassert>
-#include <cstdio>
 
 #include <com/sun/star/awt/WindowAttribute.hpp>
 #include <com/sun/star/awt/XMessageBox.hpp>
@@ -106,7 +105,7 @@ void ShowMessageBox( uno::Reference< lang::XMultiServiceFactory > const& xFactor
 
 void LayoutRoot::error( OUString const& message )
 {
-    printf( "%s\n", OUSTRING_CSTR( message ) );
+    OSL_TRACE( "%s\n", OUSTRING_CSTR( message ) );
     ShowMessageBox( mxFactory, mxToolkit,
                     OUString::createFromAscii( "Fatal error" ),
                     message );
diff --git toolkit/source/layout/core/table.cxx toolkit/source/layout/core/table.cxx
index 87c15e2..2beefc1 100644
--- toolkit/source/layout/core/table.cxx
+++ toolkit/source/layout/core/table.cxx
@@ -47,20 +47,20 @@ namespace layoutimpl
 using namespace com::sun::star;
 
 Table::ChildProps::ChildProps( Table::ChildData *pData )
-    {
-        addProp( RTL_CONSTASCII_USTRINGPARAM( "XExpand" ),
-                 ::getCppuType( static_cast< const sal_Bool* >( NULL ) ),
+{
+    addProp( RTL_CONSTASCII_USTRINGPARAM( "XExpand" ),
+             ::getCppuType( static_cast< const sal_Bool* >( NULL ) ),
              &( pData->mbExpand[ 0 ] ) );
-        addProp( RTL_CONSTASCII_USTRINGPARAM( "YExpand" ),
-                 ::getCppuType( static_cast< const sal_Bool* >( NULL ) ),
+    addProp( RTL_CONSTASCII_USTRINGPARAM( "YExpand" ),
+             ::getCppuType( static_cast< const sal_Bool* >( NULL ) ),
              &( pData->mbExpand[ 1 ] ) );
-        addProp( RTL_CONSTASCII_USTRINGPARAM( "ColSpan" ),
-                 ::getCppuType( static_cast< const sal_Int32* >( NULL ) ),
+    addProp( RTL_CONSTASCII_USTRINGPARAM( "ColSpan" ),
+             ::getCppuType( static_cast< const sal_Int32* >( NULL ) ),
              &( pData->mnColSpan ) );
-        addProp( RTL_CONSTASCII_USTRINGPARAM( "RowSpan" ),
-                 ::getCppuType( static_cast< const sal_Int32* >( NULL ) ),
+    addProp( RTL_CONSTASCII_USTRINGPARAM( "RowSpan" ),
+             ::getCppuType( static_cast< const sal_Int32* >( NULL ) ),
              &( pData->mnRowSpan ) );
-    }
+}
 
 bool Table::ChildData::isVisible()
 {
@@ -79,21 +79,21 @@ Table::Table()
 
 Table::ChildData::ChildData( uno::Reference< awt::XLayoutConstrains > const& xChild )
     : Box_Base::ChildData( xChild )
-//    , mbExpand( { 0, 1 } )
+//    , mbExpand( { 1, 1 } )
     , mnColSpan( 1 )
     , mnRowSpan( 1 )
     , mnLeftCol( 0 )
     , mnRightCol( 0 )
     , mnTopRow( 0 )
     , mnBottomRow( 0 )
-    {
-    mbExpand[ 0 ] = 0;
+{
+    mbExpand[ 0 ] = 1;
     mbExpand[ 1 ] = 1;
 }
 
 Table::ChildData*
 Table::createChild( uno::Reference< awt::XLayoutConstrains > const& xChild )
-        {
+{
     return new ChildData( xChild );
 }
 
@@ -106,9 +106,9 @@ Table::createChildProps( Box_Base::ChildData *pData )
 void SAL_CALL
 Table::addChild( const uno::Reference< awt::XLayoutConstrains >& xChild )
     throw( uno::RuntimeException, awt::MaxChildrenException )
-        {
+{
     if ( xChild.is() )
-            {
+    {
         Box_Base::addChild( xChild );
         // cause of flicker
         allocateChildAt( xChild, awt::Rectangle( 0,0,0,0 ) );
@@ -240,14 +240,14 @@ Table::getMinimumSize() throw( uno::RuntimeException )
     mnColExpandables =( mnRowExpandables = 0 );
     maRequisition.Width =( maRequisition.Height = 0 );
     for ( std::vector<GroupData>::iterator it = maCols.begin();
-         it != maCols.end(); it++ )
+          it != maCols.end(); it++ )
     {
         maRequisition.Width += it->mnSize;
         if ( it->mbExpand )
             mnColExpandables++;
     }
     for ( std::vector<GroupData>::iterator it = maRows.begin();
-         it != maRows.end(); it++ )
+          it != maRows.end(); it++ )
     {
         maRequisition.Height += it->mnSize;
         if ( it->mbExpand )
diff --git toolkit/source/layout/core/table.hxx toolkit/source/layout/core/table.hxx
index dac2aaa..b41be25 100644
--- toolkit/source/layout/core/table.hxx
+++ toolkit/source/layout/core/table.hxx
@@ -99,8 +99,6 @@ public:
     virtual sal_Int32 SAL_CALL getHeightForWidth( sal_Int32 /*nWidth*/ )
     throw(css::uno::RuntimeException)
     { return maRequisition.Height; }
-
-    PROPHELPER_SET_INFO
 };
 
 } //  namespace layoutimpl
diff --git toolkit/source/layout/core/translate.cxx toolkit/source/layout/core/translate.cxx
index d0fc9f7..b059984 100644
--- toolkit/source/layout/core/translate.cxx
+++ toolkit/source/layout/core/translate.cxx
@@ -58,13 +58,13 @@ getLocaleSubdirList( lang::Locale const& rLocale )
 {
     std::list<OUString> aSubdirs;
     aSubdirs.push_front( OUString::createFromAscii( "." ) );
-    aSubdirs.push_front( OUString::createFromAscii( "en_US" ) );
+    aSubdirs.push_front( OUString::createFromAscii( "en-US" ) );
     if ( rLocale.Language.getLength() )
         aSubdirs.push_front( rLocale.Language );
     if ( rLocale.Country.getLength() )
     {
         OUString aLocaleCountry = rLocale.Language
-            + OUString::createFromAscii( "_" )
+            + OUString::createFromAscii( "-" )
             + rLocale.Country;
         aSubdirs.push_front( aLocaleCountry );
         if ( rLocale.Variant.getLength() )
diff --git toolkit/src2xml/source/src2xml.py toolkit/src2xml/source/src2xml.py
index 128cfa1..b03c5e6 100644
--- toolkit/src2xml/source/src2xml.py
+++ toolkit/src2xml/source/src2xml.py
@@ -182,7 +182,7 @@ def post_process (s):
     s = s.replace ('<modaldialog', '<modaldialog sizeable="true"')
     s = s.replace (' rid=', ' id=')
     s = s.replace (' border="true"', ' has_border="true"')
-    s = s.replace (' def-button="true"', ' default="true"')
+    s = s.replace (' def-button="true"', ' defbutton="true"')
     s = s.replace (' drop-down="', ' dropdown="')
     s = s.replace (' tab-stop="', ' tabstop="')
     return s
diff --git toolkit/uiconfig/layout/tab-dialog.xml toolkit/uiconfig/layout/tab-dialog.xml
index 1d631e9..2639752 100644
--- toolkit/uiconfig/layout/tab-dialog.xml
+++ toolkit/uiconfig/layout/tab-dialog.xml
@@ -9,7 +9,7 @@
 	<tabcontrol id="ID_TABCONTROL"/>
 	<dialogbuttonhbox border="5" spacing="5" cnt:expand="false">
 	    <flow/>
-	    <okbutton id="BTN_OK" show="false"/>
+	    <okbutton defbutton="true" id="BTN_OK" show="false"/>
 	    <pushbutton id="BTN_USER" _label="&lt;User&gt;" show="false"/>
 	    <cancelbutton id="BTN_CANCEL" show="false"/>
 	    <helpbutton id="BTN_HELP" show="false"/>
diff --git toolkit/workben/layout/TEST toolkit/workben/layout/TEST
index d6a20cb..8a04a66 100755
--- toolkit/workben/layout/TEST
+++ toolkit/workben/layout/TEST
@@ -1,3 +1,11 @@
-export OOO_INSTALL_PREFIX=$(cd ../../../../.. && pwd)/ooo-svn
+#! /bin/sh
+for d in $(cd ../../../../.. && pwd)/ooo-svn /usr/lib/ooo3/; do
+    if test -d $d; then
+        export OOO_INSTALL_PREFIX=$d
+	break
+    fi
+done
+export SAL_ALLOW_LINKOO_SYMLINKS=1
 test -d $OOO_INSTALL_PREFIX/openoffice.org3 && export OOO_PREFIX_BRAND=$OOO_INSTALL_PREFIX/openoffice.org3 || export OOO_PREFIX_BRAND=$OOO_INSTALL_PREFIX
-LD_LIBRARY_PATH=$OOO_PREFIX_BRAND/basis-link/program:$OOO_PREFIX_BRAND/basis-link/ure-link/lib ../../unxlngx6.pro/bin/test "$@"
+export LD_LIBRARY_PATH="$OOO_PREFIX_BRAND/basis-link/program:$OOO_PREFIX_BRAND/basis-link/ure-link/lib"
+../../unx*.pro/bin/test "$@"
diff --git toolkit/workben/layout/align-test.xml toolkit/workben/layout/align-test.xml
new file mode 100644
index 0000000..f01830e
--- /dev/null
+++ toolkit/workben/layout/align-test.xml
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- This is a template.  i18n translation is not performed in-place;
+     i18n translated xml files are generated from this template by
+     transex3/layout/tralay.  !-->
+
+<dialog xmlns="http://openoffice.org/2007/layout"
+        xmlns:cnt="http://openoffice.org/2007/layout/container"
+        title="Align Test" optimumsize="true" has_border="true" sizeable="true" moveable="true">
+    <vbox spacing="5" border="5">
+	<fixedline/>
+	<hbox spacing="0" border="0">
+	    <vbox spacing="5" border="5">
+		<fixedline _text="X" cnt:expand="false"/>
+		<pushbutton label="foo"/>
+		<pushbutton label="foo"/>
+		<pushbutton label="foo"/>
+		<pushbutton label="foo"/>
+		<pushbutton label="foo"/>
+		<pushbutton label="foo"/>
+		<pushbutton label="foo"/>
+	    </vbox>
+	    <vbox spacing="5" border="5">
+		<fixedline _text="X" cnt:expand="false"/>
+		<pushbutton label="bar"/>
+	    </vbox>
+	</hbox>
+        <fixedline/>
+	<min-size cnt:spacing="0" min-width="200"/>
+    </vbox>
+</dialog>
diff --git toolkit/workben/layout/boxtest.xml toolkit/workben/layout/boxtest.xml
index 8015b3f..263ae40 100644
--- toolkit/workben/layout/boxtest.xml
+++ toolkit/workben/layout/boxtest.xml
@@ -4,6 +4,8 @@
         xmlns:cnt="http://openoffice.org/2007/layout/container"
         title="Box Test" optimumsize="true" has_border="true" sizeable="true" moveable="true">
 	<vbox>
+	  <table>
+	    <vbox>
 		<hbox cnt:expand="false" cnt:fill="false">
 			<pushbutton label="XML Left"
 			            cnt:expand="false" cnt:fill="false"/>
@@ -28,5 +30,7 @@
 	                <pushbutton label="XML Right"
 	                            cnt:expand="true" cnt:fill="true"/>
 	        </hbox>
+		</vbox>
+	</table>
 	</vbox>
 </dialog>
diff --git toolkit/workben/layout/editor.cxx toolkit/workben/layout/editor.cxx
index f3ad65c..e10e54e 100644
--- toolkit/workben/layout/editor.cxx
+++ toolkit/workben/layout/editor.cxx
@@ -344,9 +344,14 @@ public:
             {
                 beans::Property prop = it.next();
                 rtl::OUString name( prop.Name );
-                rtl::OUString value( pChild->getProperty( name, CONTAINER_PROPERTY ) );
-                std::pair< rtl::OUString, rtl::OUString > pair( name, value );
-                pChild->maOriChildProps.push_back( pair );
+		try {
+			rtl::OUString value( pChild->getProperty( name, CONTAINER_PROPERTY ) );
+			std::pair< rtl::OUString, rtl::OUString > pair( name, value );
+			pChild->maOriChildProps.push_back( pair );
+		} catch ( beans::UnknownPropertyException &rEx ) {
+			fprintf (stderr, "ERROR: widget reports that it has a property it cannot return: '%s' this normally means that someone screwed up their PROPERTY_SET_INFO macro usage.\n",
+				 rtl::OUStringToOString (rEx.Message, RTL_TEXTENCODING_UTF8).getStr());
+		}
             }
         }
 
diff --git toolkit/workben/layout/empty.xml toolkit/workben/layout/empty.xml
new file mode 100644
index 0000000..a8408e5
--- /dev/null
+++ toolkit/workben/layout/empty.xml
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- This is a template.  i18n translation is not performed in-place;
+     i18n translated xml files are generated from this template by
+     transex3/layout/tralay.  !-->
+
+<dialog xmlns="http://openoffice.org/2007/layout"
+        xmlns:cnt="http://openoffice.org/2007/layout/container"
+        title="Empty Test" optimumsize="true" has_border="true" sizeable="true" moveable="true">
+  <vbox spacing="1" border="10">
+    <min-size min-width="200"/>
+    <fixedline height="1"/>
+    <vbox spacing="10">
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+      <vbox show="false"/>
+    </vbox>
+    <fixedline height="1"/>
+  </vbox>
+</dialog>
diff --git toolkit/workben/layout/insert-sheet.xml toolkit/workben/layout/insert-sheet.xml
index 0ce5843..517a24d 100644
--- toolkit/workben/layout/insert-sheet.xml
+++ toolkit/workben/layout/insert-sheet.xml
@@ -49,7 +49,7 @@
 	    </vbox>
 	</hbox>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK"/>
+            <okbutton defbutton="true" id="BTN_OK"/>
             <cancelbutton id="BTN_CANCEL"/>
             <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/move-copy-sheet.xml toolkit/workben/layout/move-copy-sheet.xml
index 79720df..6e68b5a 100644
--- toolkit/workben/layout/move-copy-sheet.xml
+++ toolkit/workben/layout/move-copy-sheet.xml
@@ -15,7 +15,7 @@
 	    <flow/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
diff --git toolkit/workben/layout/plugin.xml toolkit/workben/layout/plugin.xml
index f630e72..908907d 100644
--- toolkit/workben/layout/plugin.xml
+++ toolkit/workben/layout/plugin.xml
@@ -20,7 +20,7 @@
 	</hbox>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
-	    <okbutton id="BTN_OK"/>
+	    <okbutton defbutton="true" id="BTN_OK"/>
 	    <cancelbutton id="BTN_CANCEL" />
 	    <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/recover.xml toolkit/workben/layout/recover.xml
index 7c96191..1d74b59 100644
--- toolkit/workben/layout/recover.xml
+++ toolkit/workben/layout/recover.xml
@@ -58,7 +58,7 @@ LINE TEST"/>
 	<checkbox id="CH_LOGFILE" _label="Open additional text document to display the restore logfile"/>
 	<dialogbuttonhbox border="5" spacing="5">
 	    <flow/>
-	    <okbutton id="BTN_OK" _label="Restore"/>
+	    <okbutton defbutton="true" id="BTN_OK" _label="Restore"/>
 	    <cancelbutton id="BTN_CANCEL" />
 	    <helpbutton id="BTN_HELP"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/sortdlg.xml toolkit/workben/layout/sortdlg.xml
index a17920e..ce45a60 100644
--- toolkit/workben/layout/sortdlg.xml
+++ toolkit/workben/layout/sortdlg.xml
@@ -56,7 +56,7 @@
             </pagelist>
         </tabcontrol>
         <hbox>
-            <okbutton id="1" tabstop="true"/>
+            <okbutton defbutton="true" id="1" tabstop="true"/>
             <cancelbutton id="1" tabstop="true"/>
             <helpbutton id="1" tabstop="true"/>
             <pushbutton id="1" tabstop="true"/>
diff --git toolkit/workben/layout/string-input.xml toolkit/workben/layout/string-input.xml
index 0a59899..a29a564 100644
--- toolkit/workben/layout/string-input.xml
+++ toolkit/workben/layout/string-input.xml
@@ -8,7 +8,7 @@
         <fixedtext id="FT_LABEL" label="FT-LABEL"/>
         <edit has_border="true" id="ED_INPUT" tabstop="true"/>
 	<dialogbuttonhbox border="5" spacing="5">
-            <okbutton default="true" id="BTN_OK" tabstop="true"/>
+            <okbutton defbutton="true" id="BTN_OK" tabstop="true"/>
             <cancelbutton id="BTN_CANCEL" tabstop="true"/>
             <helpbutton id="BTN_HELP" tabstop="true"/>
 	</dialogbuttonhbox>
diff --git toolkit/workben/layout/tab-dialog.xml toolkit/workben/layout/tab-dialog.xml
index 1d631e9..2639752 100644
--- toolkit/workben/layout/tab-dialog.xml
+++ toolkit/workben/layout/tab-dialog.xml
@@ -9,7 +9,7 @@
 	<tabcontrol id="ID_TABCONTROL"/>
 	<dialogbuttonhbox border="5" spacing="5" cnt:expand="false">
 	    <flow/>
-	    <okbutton id="BTN_OK" show="false"/>
+	    <okbutton defbutton="true" id="BTN_OK" show="false"/>
 	    <pushbutton id="BTN_USER" _label="&lt;User&gt;" show="false"/>
 	    <cancelbutton id="BTN_CANCEL" show="false"/>
 	    <helpbutton id="BTN_HELP" show="false"/>
diff --git toolkit/workben/layout/wordcount-plain.xml toolkit/workben/layout/wordcount-plain.xml
index cf31a76..a7c34fe 100644
--- toolkit/workben/layout/wordcount-plain.xml
+++ toolkit/workben/layout/wordcount-plain.xml
@@ -22,7 +22,7 @@
         </hbox>
         <fixedline height="8" id="FL_BOTTOM" width="158" x="6" y="77"/>
         <hbox>
-            <okbutton def-button="true" height="14" id="PB_OK" width="50" x="61" y="88"/>
+            <okbutton defbutton="true" height="14" id="PB_OK" width="50" x="61" y="88"/>
             <helpbutton height="14" id="PB_HELP" width="50" x="114" y="88"/>
         </hbox>
     </vbox>
diff --git toolkit/workben/layout/wordcount-tight.xml toolkit/workben/layout/wordcount-tight.xml
index 9ff84b3..b027510 100644
--- toolkit/workben/layout/wordcount-tight.xml
+++ toolkit/workben/layout/wordcount-tight.xml
@@ -28,7 +28,7 @@
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
         <hbox border="5">
 	    <flow homogeneous="true"/>
-	    <okbutton id="PB_OK" cnt:expand="false" default="true"/>
+	    <okbutton id="PB_OK" cnt:expand="false" defbutton="true"/>
 	    <helpbutton id="PB_HELP" cnt:expand="false"/>
         </hbox>
     </vbox>
diff --git toolkit/workben/layout/wordcount.xml toolkit/workben/layout/wordcount.xml
index e922294..fc7ff94 100644
--- toolkit/workben/layout/wordcount.xml
+++ toolkit/workben/layout/wordcount.xml
@@ -9,10 +9,10 @@
              id="DLG_WORDCOUNT"
              sizeable="true" moveable="true" optimumsize="true" closeable="true"
              has_border="true" sv-look="true" _title="Word Count">
-    <vbox border="5" spacing="5">
+    <vbox border="12" spacing="6">
         <fixedline id="FL_CURRENT" _text="Current selection"/>
 	<hbox>
-	    <vbox cnt:padding="13" spacing="5">
+	    <vbox cnt:padding="12" spacing="6">
 		<hbox>
 		    <fixedtext id="FT_CURRENTWORD" _label="Words:"/>
 		    <flow cnt:padding="60"/>
@@ -25,11 +25,11 @@
 		</hbox>
 	    </vbox>
 	</hbox>
-	<!-- if we could do cnt:cnt:padding="-13" here, we would not have to close
+	<!-- if we could do cnt:cnt:padding="-12" here, we would not have to close
 	     and reopen the hbox/vbox thing !-->
         <fixedline id="FL_DOC" _text="Whole document"/>
 	<hbox>
-	    <vbox cnt:padding="13" spacing="5">
+	    <vbox cnt:padding="12" spacing="6">
 		<hbox>
 		    <fixedtext id="FT_DOCWORD" _label="Words:"/>
 		    <fixedtext id="FI_DOCWORD" right="true" _label="000"/>
@@ -41,9 +41,9 @@
 	    </vbox>
 	</hbox>
         <fixedline cnt:padding="1" id="FL_BOTTOM"/>
-	<dialogbuttonhbox spacing="5">
+	<dialogbuttonhbox spacing="12">
             <flow/>
-            <okbutton id="PB_OK"/>
+            <okbutton defbutton="true" id="PB_OK"/>
             <helpbutton id="PB_HELP"/>
         </dialogbuttonhbox>
     </vbox>
diff --git toolkit/workben/layout/zoom-indent.xml toolkit/workben/layout/zoom-indent.xml
index 5a6a226..648da86 100644
--- toolkit/workben/layout/zoom-indent.xml
+++ toolkit/workben/layout/zoom-indent.xml
@@ -33,7 +33,7 @@
 	    </hbox>
 	</vbox>
 	<vbox border="6" spacing="4">
-	    <okbutton     id="OkButton"     cnt:expand="false" default="true"/>
+	    <okbutton     id="OkButton"     cnt:expand="false" defbutton="true"/>
 	    <cancelbutton id="CancelButton" cnt:expand="false"/>
 	    <helpbutton   id="HelpButton"   cnt:expand="false"/>
 	</vbox>
diff --git toolkit/workben/layout/zoom-plain.xml toolkit/workben/layout/zoom-plain.xml
index 30b72a3..c8383b8 100644
--- toolkit/workben/layout/zoom-plain.xml
+++ toolkit/workben/layout/zoom-plain.xml
@@ -3,7 +3,7 @@
 <modaldialog sizeable="true" height="142" help-id="SID_ATTR_ZOOM" moveable="true" output-size="true" id="RID_SVXDLG_ZOOM" sv-look="true" text="Zoom" width="160" xmlns="http://openoffice.org/2007/layout" xmlns:cnt="http://openoffice.org/2007/layout/container">
     <vbox>
         <fixedline height="8" id="FL_ZOOM" text="Zoom factor" width="92" x="6" y="3"/>
-        <okbutton def-button="true" height="14" id="BTN_ZOOM_OK" width="50" x="104" y="6"/>
+        <okbutton defbutton="true" height="14" id="BTN_ZOOM_OK" width="50" x="104" y="6"/>
         <radiobutton height="10" id="BTN_WHOLE_PAGE" width="80" x="12" y="14"/>
         <cancelbutton height="14" id="BTN_ZOOM_CANCEL" width="50" x="104" y="23"/>
         <radiobutton height="10" id="BTN_PAGE_WIDTH" width="80" x="12" y="27"/>
diff --git toolkit/workben/layout/zoom.xml toolkit/workben/layout/zoom.xml
index e228ace..4b4270f 100644
--- toolkit/workben/layout/zoom.xml
+++ toolkit/workben/layout/zoom.xml
@@ -8,10 +8,10 @@
              id="RID_SVXDLG_ZOOM" _title="Zoom &amp; View Layout" optimumsize="true"
 	     help-id="SID_ATTR_ZOOM"
              has_border="true" sizeable="true" moveable="true" closeable="true">
-    <vbox spacing="5" border="5">
+    <vbox spacing="0" border="12">
 	<hbox spacing="0" border="0">
-	    <vbox spacing="5" border="5">
-		<fixedline id="FL_ZOOM" _text="Zoom factor" cnt:expand="true"/>
+	    <vbox spacing="6" border="0">
+		<fixedline id="FL_ZOOM" _text="Zoom factor" cnt:expand="false"/>
 		<radiobutton radiogroup="zoom-factor" id="BTN_OPTIMAL"   _label="Optimal"/>
 		<radiobutton radiogroup="zoom-factor" id="BTN_WHOLE_PAGE" _label="Whole Page"/>
 		<radiobutton radiogroup="zoom-factor" id="BTN_PAGE_WIDTH" _label="Page Width"/>
@@ -20,7 +20,7 @@
 		    <align cnt:expand="false" cnt:fill="true">
 			<radiobutton cnt:v-align="0.5" cnt:v-fill="0" radiogroup="zoom-factor" id="BTN_USER" _label="Variable"/>
 		    </align>
-		    <flow cnt:padding="10" cnt:expand="false"/>
+		    <flow cnt:padding="12" cnt:expand="false"/>
 		    <metricfield id="ED_USER" value-step="1"
 				 repeat="true" has_border="true" spin="true"
 				 _text="100%" unit="11" custom-unit-text="%"
@@ -28,15 +28,16 @@
 				 cnt:expand="false"/>
 		</hbox>
 	    </vbox>
-	    <vbox spacing="5" border="5">
-		<fixedline id="FL_VIEWLAYOUT" _text="View layout" cnt:expand="true"/>
+	    <flow/>
+	    <vbox spacing="6" border="0">
+		<fixedline id="FL_VIEWLAYOUT" _text="View layout" cnt:expand="false"/>
 		<radiobutton radiogroup="zoom-layout" id="BTN_AUTOMATIC" _label="~Automatic"/>
 		<radiobutton radiogroup="zoom-layout" id="BTN_SINGLE"    _label="~Single page"/>
 		<hbox cnt:expand="false" cnt:fill="true">
 		    <align cnt:expand="false" cnt:fill="true">
 			<radiobutton cnt:v-align="0.5" cnt:v-fill="0" radiogroup="zoom-layout" id="BTN_COLUMNS" _label="~Columns"/>
 		    </align>
-		    <flow cnt:padding="10" cnt:expand="false"/>
+		    <flow cnt:padding="12" cnt:expand="false"/>
 		    <metricfield id="ED_COLUMNS" value-step="1"
 				 repeat="true" has_border="true" spin="true"
 				 _text="100%"
@@ -45,15 +46,15 @@
 				 spin-value="1"
 				 cnt:expand="false"/>
 		</hbox>
-		<hbox><flow cnt:expand="false" cnt:padding="10"/><checkbox id="CHK_BOOK" _label="~Book mode"/></hbox>
+		<hbox><flow cnt:expand="false" cnt:padding="12"/><checkbox id="CHK_BOOK" _label="~Book mode"/></hbox>
 	    </vbox>
 	</hbox>
-	<fixedline cnt:padding="1" id="FL_BOTTOM"/>
-	<dialogbuttonhbox border="5" spacing="5">
+        <fixedline cnt:padding="6" id="FL_BOTTOM"/>
+	<dialogbuttonhbox border="0" spacing="12">
 	    <flow/>
-	    <okbutton     id="BTN_ZOOM_OK"/>
+	    <okbutton defbutton="true" id="BTN_ZOOM_OK"/>
 	    <cancelbutton id="BTN_ZOOM_CANCEL"/>
-	    <helpbutton   id="BTN_ZOOM_HELP"/>
+	    <helpbutton id="BTN_ZOOM_HELP"/>
 	</dialogbuttonhbox>
     </vbox>
 </modaldialog>
